### DISCUSSION TITLE
Idea: Cleaner startpage

### DISCUSSION THREAD

**Discussion by tooomm** [Senryoku/Draftmancer#247] [Ideas]
When a new user (or even a veteran) join the page and want to play some games, the UI is very comprehensive currently.
There is a lot of text and many things want your attention. It takes a while to get used to it and get comfortable. 

I took a look at how the overwhelming initial impression can get streamlined to focus on the most important basics and be more supportive without changing too much and interrupting the known experience and build on what users know already.
Came up with a mock-up after this steps:

- Find a more suitable location for specific explanations that are only relevant in certain situations (optimize for the vast majority)
    - Move collection information next to their context/settings
      Also, in my experience within various communities, users mostly don't upload collections
![collection import](https://user-images.githubusercontent.com/9874850/114389183-f9012900-9b94-11eb-83ba-c7c707c6a394.png)

    - Basic setup information for new users to get them started are ideally not be needed if the UI is intuitive, something tricky to archive. On the other hand, there should be some help for new users somewhere!
However, the fundamental steps on how to use the page are very prominent (yet barely read imo) currently.
I opted to move them to their own modal and gave them the most prominent location in the help section. The link could even be stressed more with an icon or other text formatting.
![get started](https://user-images.githubusercontent.com/9874850/114389306-20f08c80-9b95-11eb-9d64-2ceeef0c13ab.png)
(only a pic of an interim step to understand the process better)

- Restructure elements a bit to stress important information more (e.g. news to the left to recognize updates immediatly, supportive panel and tools always at the same right location with a different background, relocate public games, everything a bit more centered with more clearance)
![layout](https://user-images.githubusercontent.com/9874850/114389585-75940780-9b95-11eb-8504-bae74998bedc.png)

The head and tag line could be adjusted as well to better match the new layout.
This is only a quick draft to show a potential route of course.

- A Discord and GitHub icon (font awesome) with links to the community/repo in the footer would be nice, too
![community icons](https://user-images.githubusercontent.com/9874850/114389896-dcb1bc00-9b95-11eb-8584-2456684f2ed5.png)
The news section could probably be thinned out a bit as well.

I'm more than happy to receive feedback and other suggestions!

<br>

---

<br>

**Comparison**

Old:
![before](https://user-images.githubusercontent.com/9874850/114390299-5c3f8b00-9b96-11eb-8603-2f7e7c252643.png)

New (Mockup):
![layout2](https://user-images.githubusercontent.com/9874850/114389593-76c53480-9b95-11eb-9e91-b61fd4d2b259.png)

**Comment by Senryoku** [comment]
Implemented some of the proposed changes here https://github.com/Senryoku/MTGADraft/pull/248 ; Will experiment more later.

**Comment by tooomm** [comment]
## To document state and progress of #248 (Merged):

**First iteration:**
- Move `Basic Setup` and `Collection Import` widgets to own modals
- Updates in `Help` widget
- Reorder widgets layout
- Updates in `Footer`

![page](https://user-images.githubusercontent.com/9874850/114547914-56fa4300-9c5f-11eb-8287-c88f498c4526.png)

<details>
<summary>Modals</summary>

![modal1](https://user-images.githubusercontent.com/9874850/114549440-4fd43480-9c61-11eb-967f-a25a701b7347.png)
![modal2](https://user-images.githubusercontent.com/9874850/114549446-519df800-9c61-11eb-8faf-f4a28f1e3372.png)

</details>

---

**Update:**
- More widgets layout updates
- Updates in `Footer`
- Updates in `Modals`

![page](https://user-images.githubusercontent.com/9874850/114706454-44e4d700-9d29-11eb-9bda-f16941a56d6f.png)

<details>
<summary>Modals</summary>

![modal1](https://user-images.githubusercontent.com/9874850/114706437-40202300-9d29-11eb-8ad2-193f82420ade.png)
![modal2](https://user-images.githubusercontent.com/9874850/114706443-41515000-9d29-11eb-9251-3313da9193e7.png)

</details>

---

**Update2:**
- Updates in `Help` and `Tools` widget
- Updates in `Footer`

![page](https://user-images.githubusercontent.com/9874850/115053977-ff1e3f00-9edf-11eb-8157-7ad5da0c7477.png)

*Modals unchanged.*

---

**Final state:**
- Update buttons in settings at top

![page](https://user-images.githubusercontent.com/9874850/115119319-04df5780-9fa8-11eb-8e86-f06120316c13.png)

*Modals unchanged.*

**Comment by tooomm** [comment]
Step 1 is finished. :)
Looking forward to the initial feedback people might drop about it. I think it's a very solid start.

As we realized in the PR when discussing ideas and other solutions, the crowded settings panel on top are a concern and lead to problems for further improvements.

What do you think about this?
The setting for the draft timer is rarely adjusted and a kinda specific setting - it should be where the other advanced and detailed settings are... in the `Additional Session Settings` modal (`Draft Specific Settings` category). The settings then still have the most basic and important settings to start a game quickly.
Then there is only the Bot option left which we can easily find a nice alternative place for:
- currently:
![settings cleanup-old](https://user-images.githubusercontent.com/9874850/115119731-35c08c00-9faa-11eb-8024-ef8e3ea31d64.png)
- mockup:
![settings cleanup-new](https://user-images.githubusercontent.com/9874850/115119749-496bf280-9faa-11eb-9f6d-bada6ce5606c.png)
Instead the `[ + ]` button (probably a `[ - ]` one would be needed as well), there are other options like using the [`user-plus`](https://fontawesome.com/icons/user-plus?style=solid) (and -minus) icon.<br>
Another option I like is, using the [`user-astronaut`](https://fontawesome.com/icons/user-astronaut?style=solid) or [`robot`](https://fontawesome.com/icons/robot?style=solid) icon and keep the same input box for numbers and increment/decrement option as currently:
![bots](https://user-images.githubusercontent.com/9874850/115120015-71a82100-9fab-11eb-9986-f70f811b8e4e.png)


The input fields for `User Name` and `Session ID` (especially the last one) could probably be smaller without causing issues as well.
The Session ID one can display crazy 24/25 digits at once before starting to cut off for me.

**Comment by tooomm** [comment]
Another simple improvement:

The enable/disable notifications check box could be an icon toggle and save space - like the enable/disable sound one:
![notification](https://user-images.githubusercontent.com/9874850/116527292-cff1cf80-a8da-11eb-91a8-d02a934ffcb6.png)
![notifications-icon](https://user-images.githubusercontent.com/9874850/116528956-b3569700-a8dc-11eb-9f64-47affc367a2c.png)

The bell icon is widely used for notifications in apps and on webpages: [bell](https://fontawesome.com/icons/bell?style=solid) and [bell-slash](https://fontawesome.com/icons/bell-slash?style=solid)
![notifications](https://user-images.githubusercontent.com/9874850/116534214-89a06e80-a8e2-11eb-9b91-7135d82617d5.png)

**Reply by Senryoku** [reply]
Yes, that's better f8c4d9d4f82a1837598f841556a9762fb2ca5395
